### ISSUE TITLE
develop a autoencoder for pretraining a CNN layer

### ISSUE THREAD

**Issue by sumo8291** [fchollet/keras#252]
I am trying to use the autoencoder for pretraining the 1st conv layer of CNN.

The code is as below:

model = Sequential()
    #model.add(Convolution2D(24, 1,7,7, border_mode='valid')) #42x42x24
    model.add(AutoEncoder(encoder=Convolution2D(24, 1,7,7, border_mode='valid'), decoder=Convolution2D(1, 24,7,7, border_mode='full'), 
    	output_reconstruction=False, tie_weights=True))

    opt=RMSprop()
    ### Compile the loss
    model.compile(loss='mean_squared_error', optimizer=opt)
    # Training Neural Network
    model.fit(X_train, X_train, batch_size=batch_size, nb_epoch=nb_epoch, show_accuracy=True, verbose=2)

#where X_train is data of 28709.

The error i am getting is as follows:
![screen shot 2015-06-20 at 1 44 36 pm](https://cloud.githubusercontent.com/assets/8149573/8266558/d81ca138-1751-11e5-918c-fd794173255a.png)

Can you please help in how to remove the error?

**Comment by valexandersaulys** [comment]
Hey did you have any luck on this? I'm curious how you set the weights of the final model with the ones from the pretrained autoencoder.

**Comment by danosaku** [comment]
You need reshape your tensor. Use e.g (28709, 1, 48, 48), where the arguments are the number of data, number of channels, width and height image, respectively.
